doc/mgmt/sonic_stretch_management_vrf_design.md

```diff
@@ -0,0 +1,400 @@
+broadcast 10.16.210.255 dev eth0 proto kernel scope link src 10.16.210.75 
+broadcast 127.0.0.0 dev lo-m proto kernel scope link src 127.0.0.1 
+127.0.0.0/8 dev lo-m proto kernel scope link src 127.0.0.1 
+local 127.0.0.1 dev lo-m proto kernel scope host src 127.0.0.1 
+broadcast 127.255.255.255 dev lo-m proto kernel scope link src 127.0.0.1 
+root@sonic:/etc/init.d#
+```
+
+
+```
+root@sonic:/etc/init.d# show management_interface address 
+Management IP address = 10.16.210.75/24
+Management NetWork Default Gateway = 10.16.210.254
+root@sonic:/etc/init.d#
+```
+
+These CLI commands are implemented as part of the pull request [PR463](https://github.com/Azure/sonic-utilities/pull/463)
+
+### IP Application Design
+This section explains the behavior of each application on the default VRF and management VRF. Application functionality differs based on whether the application is used to connect to the application daemons running in the device or the application is triggered from the device.
+
+#### Application Daemons In The Device  
+##### TCP Application Daemons  
+Linux kernel has got the flag tcp_l3mdev_accept which is required for enabling the TCP applications to receive and process packets from both data VRF and management VRF. By default, this flag is already enabled. With this flag enabled, TCP is capable of accepting the incoming connections that internally binds the incoming interface to the connection. i.e. When connection request arrives via the front-panel ports, TCP accepts the connection via the front-panel port; when connect request arrives via the management port (eth0), it accepts via the management port. Example: sshd, ftpd, etc., No changes are required in these deamons to make them work on both management VRF and default VRF.
+
+##### UDP Application Daemons
+Linux 4.9 does not support `udp_l3mdev_accept`; corresponding patch is back ported to SONiC Linux 4.9. With this patch, UDP applications will be able to receive and process packets from both data VRF and management VRF. UDP being a connectionless protocol, these is no concept of accepting the connection through the same incoming port (VRF) and hence UDP applications will not be able to specify the interface through which the reply has to be sent. i.e. When UDP applications send a reply reply packet, there is no reference to the port in which the request packet had arrived. Hence the Linux stack will try to route this packet via default routing table, even if request packet had arrived via management port. Such UDP application daemon code needs to be modified to use the incoming port reference and use it for sending replies via the same VRF through which the request had arrived.
+
+##### IP Application Daemons
+Most IP applications (like PING) always have the incoming port/VRF association with the reply packet and hence they will continue to work without any modification. 
+
+#### Applications Originating From the Device

```

Comment:
Will there be any description regarding "docker pull" when Management VRF is enabled? Currently the cgexec does not seem to work for it, as it is doing all its lookups, and fetches (DNS, HTTPS) using the default VRF. The current workaround seems to be by  adding a rule to the routing policy database to lookup the Management VRF, if route cannot be found. Thinking out loud here, perhaps an option inside the MGMT_VRF_CONFIG table to allow for this?

Reply:
No. There wont be any description for "docker pull" since the original scope during development was for "snmp" and "ntp" only. 
Once if this is merged, any contributor is welcome to enhance the design and the code.